tests/unit/CoreListTest.php

```diff
@@ -0,0 +1,373 @@
+<?php
+
+namespace WPGraphQL\ContentBlocks\Unit;
+
+final class CoreListTest extends PluginTestCase {
+	/**
+	 * The ID of the post created for the test.
+	 *
+	 * @var int
+	 */
+	public $post_id;
+
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->post_id = wp_insert_post(
+			[
+				'post_title'   => 'Post with List',
+				'post_content' => '',
+				'post_status'  => 'publish',
+			]
+		);
+
+		\WPGraphQL::clear_schema();
+	}
+
+	public function tearDown(): void {
+		parent::tearDown();
+
+		wp_delete_post( $this->post_id, true );
+
+		\WPGraphQL::clear_schema();
+	}
+
+	public function query(): string {
+		return '
+			fragment CoreListBlockFragment on CoreList {
+				attributes {
+					anchor
+					backgroundColor
+					className
+					cssClassName
+					fontFamily
+					fontSize
+					gradient
+					lock
+					# metadata
+					ordered
+					placeholder
+					reversed
+					start
+					style
+					textColor
+					type
+					values
+				}
+			}
+
+			query Post( $id: ID! ) {
+				post(id: $id, idType: DATABASE_ID) {
+					databaseId
+					editorBlocks {
+						apiVersion
+						blockEditorCategoryName
+						clientId
+						cssClassNames
+						innerBlocks {
+							name
+						}
+						isDynamic
+						name
+						parentClientId
+						renderedHtml
+						...CoreListBlockFragment
+					}
+				}
+			}
+		';
+	}
+
+	/**
+	 * Test case for retrieving core list block fields and attributes.
+	 *
+	 * The following aspects are tested:
+	 * - The absence of errors in the GraphQL response.
+	 * - Presence of the 'data' and 'post' keys in the response.
+	 * - Matching post ID.
+	 * - Correct block name ('core/list').
+	 * - Correct retrieval of the block's attributes, especially 'anchor', 'backgroundColor', 'className', and 'cssClassName'.
+	 *
+	 * @return void
+	 */
+	public function test_retrieve_core_list_fields_and_attribute() {
+		$block_content = '
+			<!-- wp:list {"className":"test-css-class-name","backgroundColor":"accent-4"} -->
+				<ul id="test-anchor" class="wp-block-list test-css-class-name has-accent-4-background-color has-background">
+					<!-- wp:list-item --><li>Truck</li><!-- /wp:list-item -->
+					<!-- wp:list-item --><li>Train</li><!-- /wp:list-item -->
+				</ul>
+			<!-- /wp:list -->
+		';
+
+		// Set post content.
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$query     = $this->query();
+		$variables = [
+			'id' => $this->post_id,
+		];
+
+		$actual = graphql( compact( 'query', 'variables' ) );
+
+		$this->assertArrayNotHasKey( 'errors', $actual, 'There should not be any errors' );
+		$this->assertArrayHasKey( 'data', $actual, 'The data key should be present' );
+		$this->assertArrayHasKey( 'post', $actual['data'], 'The post key should be present' );
+		$this->assertEquals( $this->post_id, $actual['data']['post']['databaseId'], 'The post ID should match' );
```

Comment:
```suggestion
		$this->assertEquals( $this->post_id, $actual['data']['post']['databaseId'], 'The post ID should match' );

		$this->assertCount( 3, $actual['data']['post']['editorBlocks'] );

```

Reply:
You're also mising other basic field tests. See https://github.com/rtCamp/wp-graphql-content-blocks/blob/530233b05c3c0873d262ccee76cae650190610d7/tests/unit/CoreHeadingTest.php#L96-L118